docs/07-faq/023-kubernetes-support.md

```diff
@@ -0,0 +1,16 @@
+---
+title: Does Wing support Kubernetes?
+id: kubernetes-support
+keywords: [faq, supported provisioning engines, winglang, Wing programming language, Wing language, Kubernetes, K8S]
+---
+
+The short answer us that we don't support Kubernetes yet, but we will in the future. 
+Our roadmap is [here](https://docs.winglang.io/status#roadmap).
+
+The longer answer is that there are a different kinds of Kubernetes support with varying levels of maturity in Wing:
+1. Creation and configuration of K8S clusters through the [K8S Terraform Provider](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs). Since Wing supports any Terraform provider, it should work, but we habe not officially tested it yet.
```

Comment:
This should take only a few minutes to test, we can install `https://www.npmjs.com/package/@cdktf/provider-kubernetes` and see if it's possible to instantiate a few resources like a pod or deployment in Wing. I can write up an example tomorrow

```suggestion
1. Creation and configuration of K8S clusters through the [K8S Terraform Provider](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs). These resources can be used by installing `@cdktf/provider-kubernetes` from npm and using the resources in Wing:

(example code)
```

Reply:
Cool!

Reply:
OK - so I had a chance to play around with this, here's the example code in Wing:

```js
bring cloud;
bring "@cdktf/provider-kubernetes" as kube;

new kube.provider.KubernetesProvider();

let ns = new kube.namespace.Namespace(metadata: kube.namespace.NamespaceMetadata {
  name: "nginx",
});

let deployment = new kube.deployment.Deployment(metadata: kube.deployment.DeploymentMetadata {
  name: "nginx",
  namespace: ns.metadata.name,
}, spec: kube.deployment.DeploymentSpec {
  template: kube.deployment.DeploymentSpecTemplate {
    metadata: kube.deployment.DeploymentSpecTemplateMetadata {
      labels: {
        app: "nginx",
      },
    },
    spec: kube.deployment.DeploymentSpecTemplateSpec {
      container: [
        kube.deployment.DeploymentSpecTemplateSpecContainer {
          name: "nginx-container",
          image: "nginx",
          port: kube.deployment.DeploymentSpecTemplateSpecContainerPort {
            container_port: 80,
          },
        }
      ]
    }
  }
});
```

It produces valid terraform which is cool 👍 

But there are several ergonomic issues:
- IDE autocomplete doesn't work after you type `new kube.deployment.`
- You have to type the full type names everywhere, and it gets very cumbersome
- To compile you need to specify a cloud like `wing compile -t tf-aws` even though the template you produce isn't specific to any cloud, so it seems weird (I think we need another target like `tf` or `tf-base`?)

So maybe it's not worth showing off this capability in the FAQ yet

Reply:
> IDE autocomplete doesn't work after you type new kube.deployment.

Just wanted to add that this autocomplete bug is not specific to Kubernetes. It's on Mark's radar and should be fixed soon.
https://github.com/winglang/wing/issues/2339